### PR TITLE
Improved testing and gnome-session scripts

### PR DESCRIPTION
**Pull request by tanty**

### CODE REVIEW

**Comment by infapi00** [comment]
Uh sorry. I missed this pull request. It is still valid?

I don't have anything to say about all the stuff you created. 

So about generalizing run-n-times-whatever. I see the generalization, but not removing the current piglit and deqp ones. Is that the plan?

**Review by infapi00 — COMMENTED** [review]
It seems that you used as base the run-n-times-piglit, and probably you didn't realize that I had a run-n-times-deqp. Or that you thought that the latter didn't add anything new.

Although I like the idea of having just one script, as Im saying the deqp script has one useful feature. For each run it saved the TestResults.qpa, renaming it. This was useful in order to get the outcome of a failing test, if the flaky test tended to have more pass that failures.

**Comment by tanty** [comment]
> Uh sorry. I missed this pull request. It is still valid?

Yes, it is.

> It seems that you used as base the run-n-times-piglit, and probably you didn't realize that I had a run-n-times-deqp. Or that you thought that the latter didn't add anything new.

Oh, crap!

I just didn't see that there was another one for deqp. In any case, AFAIR, my generalization is for running deqp also through piglit so I think your other script may still be useful.

Whenever I have time I can go back to check and see if I can integrate everything in a single script.

**Comment by infapi00** [comment]
> Oh , crap!
> 
> I just didn't see that there was another one for deqp. In any case, AFAIK, my generalization is for running deqp also through piglit so I think your other script may still be useful.

Oh, then it would be good if the documentation explicitly say that. Just skimming and seeing the script doc, seems to suggest that you run vk-gl-cts directly.

Additionally, then I think that it doesn't make sense to move the script to the directory "testing", and should remain on the piglit directory.

> Whenever I have time I can go back to check and see if I can integrate everything in a single script.

Not sure if it is worth to do that, as probably there aren't too much to share.

**Comment by tanty** [comment]
> Oh, then it would be good if the documentation explicitly say that. Just skimming and seeing the script doc, seems to suggest that you run vk-gl-cts directly.

OK

> Additionally, then I think that it doesn't make sense to move the script to the directory "testing", and should remain on the piglit directory.

The thing is, that script is the only thing in that directory while in "testing" there are already plenty of scripts that actually also use piglit to do the testing. I think it made more sense to put all the testing tools together. Otherwise it seems more confusing and we should move some to piglit (?).

**Comment by infapi00** [comment]
Ok, I see what you mean. I didn't realize that piglit directory only had run-n-times for piglit. Then ok with moving it to testing, and nuking the piglit directory.

**Comment by tanty** [comment]
Updated!

File: testing/run-n-times-flaky-test.sh [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # This script runs a piglit test n times, and returns how many runs
```
Comment (infapi00) [inline testing/run-n-times-flaky-test.sh]: Now that the test is more complete, is "runs a piglit test n times" correct? Shouldn't be "runs a piglit or deqp test n times"?

**Comment by tanty** [comment]
Updated!

**Comment by tanty** [comment]
As commented through another channel, @infapi00 is OK with merging this PR but I do not have write access, so he will have to do it ...

**Comment by infapi00** [comment]
> As commented through another channel, @infapi00 is OK with merging this PR but I do not have write access, so he will have to do it ...

Argh, I missed this comment about you needing me to merge it. Sorry for that.

But I noted that some of the commits (like include meson) are merged. Would you mind to update (again) this PR? And again sorry for the mess.

**Comment by infapi00** [comment]
@tanty, I have just invited you as collaborator. I hope that would be enough to allow you to merge this changes.

**Pull request closed without merging**